### PR TITLE
thunderbird, thunderbird-bin: fix desktop item

### LINKED ISSUE #194851: thunderbird’s desktop file only supports web browser MIME types
### Describe the bug
After installing thunderbird via `environment.systemPackages`, no entries are added to Gnome Default Applications for Mail and Calendar.

### Steps To Reproduce
1. Install thunderbird
2. Open Gnome Settings
3. Go to Default Applications
4. Confirm that Thunderbird is not in Mail or Calendar lists

### Expected behavior
Thunderbird should be in Mail and Calendar in Gnome Default Applications.

### Notify maintainers

@eelco @lovesegfault @pierron @vcunat 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.68, NixOS, 22.11 (Raccoon), 22.11.20220926.62228cc`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by taku0**
Generate appropriate desktop item for Thunderbird.

Fixes #194851.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

My KDE environment somehow shows Thunderbird as a default mail reader without this PR. `xdg-open mailto:text@example.org` also starts Thunderbird without this PR. So I could not test this. Just checked generated desktop files for `thunderbird-bin` and `firefox-bin`.

### CODE REVIEW

**Comment by vcunat** [comment]
With Xfce I also already had it shown in mail readers.  This PR looks good to me, but it would be nice to have someone confirm that the GNOME issue is gone.

File: pkgs/applications/networking/browsers/firefox/wrapper.nix [hunk 1]
```diff
@@ -166,38 +166,70 @@ let
...
-          };
-        };
-      };
+        terminal = false;
+      } // (if libName == "thunderbird"
+            then {
+              genericName = "Email Client";
+              comment = "Read and write e-mails or RSS feeds, or manage tasks on calendars.";
+              categories = [
+                "Network" "Calendar" "Chat" "ContactManagement" "Email" "Feed"
```
Comment (jtojnar) [inline pkgs/applications/networking/browsers/firefox/wrapper.nix:182]: The following are printed when building the package:

```
/nix/store/0v4d62kwq4fqrhrwf470xl6vcwkgy0ry-thunderbird.desktop/share/applications/thunderbird.desktop: hint: value item "Calendar" in key "Categories" in group "Desktop Entry" can be extended with another category among the following categories: Office
/nix/store/0v4d62kwq4fqrhrwf470xl6vcwkgy0ry-thunderbird.desktop/share/applications/thunderbird.desktop: hint: value item "ContactManagement" in key "Categories" in group "Desktop Entry" can be extended with another category among the following categories: Office
```

**Review by jtojnar — COMMENTED** [review]
Can confirm `thunderbird` can now be set in GNOME Control Center:

Default applications pane of GNOME Settings with Thunderbird set as the default e-mail and calendar app

File: pkgs/applications/networking/browsers/firefox/wrapper.nix (hunk 1 above)
Comment (taku0) [reply pkgs/applications/networking/browsers/firefox/wrapper.nix:182]: Adding `Office` to `Categories` results in the following message:

```
/nix/store/q5q7kzw09sczj0kjjzcvc7lp3xs9dsnz-thunderbird.desktop/share/applications/thunderbird.desktop: hint: value "Network;Calendar;Chat;ContactManagement;Email;Feed;GTK;News;Office" for key "Categories" in group "Desktop Entry" contains more than one main category; application might appear more than once in the application menu
```

We have three options:

- A. Let Thunderbird appears under both `Network` and `Office`.
- B. Remove `Calendar` and `ContactManagement` from the category.
- C. Keep `Calendar` and `ContactManagement` without `Office`.

Comment (vcunat) [reply pkgs/applications/networking/browsers/firefox/wrapper.nix:182]: A. would be weird, I think.

**Comment by taku0** [comment]
Addressed comments.

File: pkgs/applications/networking/browsers/firefox/wrapper.nix [hunk 2]
```diff
@@ -166,38 +166,64 @@ let
...
+              mimeTypes = [
+                "message/rfc822"
+                "x-scheme-handler/mailto"
+                "text/calendar"
+                "text/x-vcard"
+              ];
+              actions = {
+                profile-manager-window = {
+                  name = "Profile Manager";
+                  exec = "${launcherName} --ProfileManger";
```
Comment (jtojnar) [inline pkgs/applications/networking/browsers/firefox/wrapper.nix:197]: Looks like this is missing a letter: Man<ins>a</ins>ger

Comment (vcunat) [reply pkgs/applications/networking/browsers/firefox/wrapper.nix:197]: Thanks.  This PR just copy&pasted the typo into another location :-)
https://github.com/NixOS/nixpkgs/pull/200823

**Pull request merged**